contracts/modules/TransferManager/VolumeDumpingRestrictionManager.sol

```diff
@@ -0,0 +1,326 @@
+pragma solidity ^0.4.24;
+
+import "./ITransferManager.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title  Transfer Manager module for restricting volume dumping within a time period for an account
+ */
+
+contract VolumeDumpingRestrictionManager is ITransferManager {
+    
+    using SafeMath for uint256;
+
+    bytes32 public constant VOLUME_RESTRICTION = "VOLUME_DUMPING_RESTRICTION";
+    
+    // permission definition
+    bytes32 public constant ADMIN = "ADMIN";
+    
+    //per volume restriction rule
+    struct VolumeRestriction {
+        uint256 percentAllowed;
+        uint256 startTime;
+        uint256 endTime;
+        uint256 rollingPeriod;
+    }
+
+    // Maps user address to a volumerestriction rule
+    mapping(address => VolumeRestriction) internal volumeRestriction;
+
+    // Maps user address to a periodId & amount transferred so far in that periodId
+    mapping(address => mapping(uint256 => uint256)) internal volumeTally;
+
+    // map user address to a list of periodIds
+    mapping(address => uint256[]) internal volumePeriodIds;
+    
+    event ModifyVolumeDumping (
+        address indexed from, 
+        uint256 percentAllowed, 
+        uint256 startTime,
+        uint256 endTime,
+        uint256 rollingPeriod
+    );
+
+    event AddNewVolumeDumping (
+        address indexed from, 
+        uint256 percentAllowed, 
+        uint256 startTime,
+        uint256 endTime,
+        uint256 rollingPeriod
+    );
+
+    event RemoveVolumeDumping (
+        address indexed from, 
+        uint256 percentAllowed, 
+        uint256 startTime,
+        uint256 endTime,
+        uint256 rollingPeriod
+
+    );
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+    public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /** @notice Used to verify the transfer transaction and prevent locked up tokens from being transferred
+     * @param _from Address of the sender
+     * @param _amount The amount of tokens to transfer
+     */
+    function verifyTransfer(address _from, address /*_to */, uint256 _amount, bytes /* _data */, bool _isTransfer) public returns(Result) {
+        
+        // function must only be called by the associated security token if _isTransfer == true
+        require(_isTransfer == false || msg.sender == securityToken, "Sender is not owner");
+
+        if(!_isTransfer) {
+            return Result.VALID;
+        }
+
+        if (!paused && _from != address(0)) {                
```

Comment:
Probably should also check that `_amount != 0` otherwise you'll add duplicate periods in `_currentAllowedAmount` I think.